### PR TITLE
Add new configuration into *Mojo: Ability to filter only file by file

### PR DESCRIPTION
**Pull request by ouaibsky**
I need to filter file by file and change the location of result.
So I update the AbstractMojo to take into account new configuration property
I take the opportunity to mark filesets property as optional.
example of usage: 

```
<configuration>
    <files>
        <singlefile>
            <sourceFile>${project.build.directory}/config/${project.artifactId}.properties</sourceFile>
            <targetFileName>${project.build.testOutputDirectory}/${project.artifactId}.properties</targetFileName>
           <filterOnlyMustacheProperties>false</filterOnlyMustacheProperties>
        </singlefile>
    </files>
</configuration>
```

### CODE REVIEW

File: src/test/java/com/xebialabs/maven/mustache/AbstractMustachifierMojoTest.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        mojo.saveSingleFile(new SingleFile("./target/test-classes/mustache/withplaceholders/config.properties", "./target/test-classes/mustache/withplaceholders/config-test.properties\\", false));
+    }
+
+    @Test
+    public void testSaveSingleFileWithFiltering() throws Exception {
+        MustachifierToValueMojo mojo = new MustachifierToValueMojo();
+        mojo.setValueSeparator(":");
+        mojo.setDelimiters("{{ }}");
+        mojo.setFilesets(new ArrayList<FileSet>());
+        mojo.saveSingleFile(new SingleFile("./target/test-classes/mustache/withplaceholders/config.properties", "./target/test-classes/mustache/withplaceholders/config-filter.properties\\", true));
```
Comment (bmoussaud) [inline src/test/java/com/xebialabs/maven/mustache/AbstractMustachifierMojoTest.java]: No assert methods

File: src/test/java/com/xebialabs/maven/mustache/AbstractMustachifierMojoTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    final static String lineBreak = System.getProperty("line.separator");
+
+    @Test
+    public void testSaveSingleFile() throws Exception {
+        MustachifierToValueMojo mojo = new MustachifierToValueMojo();
+        mojo.setValueSeparator(":");
+        mojo.setDelimiters("{{ }}");
+        mojo.setFilesets(new ArrayList<FileSet>());
+        mojo.saveSingleFile(new SingleFile("./target/test-classes/mustache/withplaceholders/config.properties", "./target/test-classes/mustache/withplaceholders/config-test.properties\\", false));
```
Comment (bmoussaud) [inline src/test/java/com/xebialabs/maven/mustache/AbstractMustachifierMojoTest.java]: No assert methods

File: src/main/java/com/xebialabs/maven/mustache/AbstractMustachifierMojo.java [hunk 3]
```diff
@@ -69,6 +82,39 @@ public void execute() throws MojoExecutionException, MojoFailureException {
...
+            parent.mkdirs();
+            save(processed, output, aFile.getModelEncoding());
+        } catch (MojoExecutionException e) {
+            getLog().error("Failed to read input file: "+input, e);
+        } catch (IOException e) {
+            getLog().warn("Failed to read input file: "+input);
+        }
+    }
+
+    final String filterMustacheOnly(final String aContent) {
```
Comment (bmoussaud) [inline src/main/java/com/xebialabs/maven/mustache/AbstractMustachifierMojo.java:109]: Filter the content only with lines that are containing the before delilmiter '{{' ?
What is the use case ? (ex example ?)

Comment (ouaibsky) [inline src/main/java/com/xebialabs/maven/mustache/AbstractMustachifierMojo.java:109]: I provide a bit  of explanation into javadoc of SingleFile#setFilterOnlyMustacheProperties

**Comment by bmoussaud** [comment]
Before merging the PR could you provide a documentation page 
- create a new file in src/site/markdown
- add the entry in src/site/site.xml

**Comment by ouaibsky** [comment]
The travis job is not working with jdk7 but I think it's not related to my changes
it's working with jdk8
Can you check please ?

**Comment by bmoussaud** [comment]
this option `<additionalparam>-Xdoclint:none</additionalparam>` in the pom file is not recognized by java 7

```
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.1:jar (attach-javadocs) on project placeholders-mustachifier-maven-plugin: MavenReportException: Error while creating archive:
[ERROR] Exit code: 1 - javadoc: error - invalid flag: -Xdoclint:none
[ERROR] 
```

**Comment by ouaibsky** [comment]
Hi
Fix has been done in order to compile with jdk 7 & 8
Can you release please ?
Thx
Christophe

**Comment by bmoussaud** [comment]
I've merged the pull request 
I'll release it in the next days
@ouaibsky  merci ;-)

**Comment by bmoussaud** [comment]
@ouaibsky 0.5 released  !

**Comment by ouaibsky** [comment]
Thx

**Pull request merged**